ruby/tzdump.rb

```diff
@@ -1,39 +1,90 @@
-require 'tzinfo'
+#!/usr/bin/env ruby
+
+## TODO:  Add some TL doc's!
+
+# To avoid having to run with `bundle exec $0`
+require "rubygems"
+require "bundler/setup"
+
+require "tzinfo"
 
 # Constants
 
-START_UTC = DateTime.new(1905, 1, 1, 0, 0, 0, '+0')
-END_UTC = DateTime.new(2035, 1, 1, 0, 0, 0, '+0')
+START_UTC = DateTime.new(1905, 1, 1, 0, 0, 0, "+0")
+END_UTC = DateTime.new(2035, 1, 1, 0, 0, 0, "+0")
 
-def format_offset(offset)
-  sign = offset < 0 ? "-" : "+"
-  offset = offset.abs
-  return '%s%02d:%02d:%02d' % [sign, offset / 3600, (offset / 60) % 60, offset % 60]
+# This is a simple wrapper for $stdout.write in-order to force CRLF line-endings for STOOUT
+module CrLf
+  attr_writer :console
+  # Over-engineered, perhaps:
+  def self.console
+    @console ||= IO.new($stdout.fileno, mode: "w", universal_newline: true)
+  end
+
+  def self.puts(*messages)
+    console.write(messages.join("\r\n"))
+    console.write("\r\n")
+  end
 end
 
-def dump_zone(console, id)
-  console.write "#{id}\r\n"
-  zone = TZInfo::Timezone.get(id)
-  period = zone.period_for_utc(START_UTC)
-  if (period.end_transition.nil? || period.end_transition.at > END_UTC)
-    console.write "Fixed: #{format_offset(period.offset.utc_total_offset)} #{period.offset.abbreviation}\r\n"
-  else
-    while !period.end_transition.nil? && period.end_transition.at < END_UTC
-      period = zone.period_for_utc(period.end_transition.at)
-      start = period.start_transition.at.to_datetime
-      formatted_start = start.strftime("%Y-%m-%dT%H:%M:%SZ")
-      formatted_offset = format_offset(period.offset.utc_total_offset)
-      console.write "#{formatted_start} #{formatted_offset} #{period.dst? ? "daylight" : "standard"} #{period.offset.abbreviation}\r\n"
+# This is a dumper class for the TZInfo data
```

Comment:
I'm actually coming at this from a C# and Java perspective, in terms of the OO. It seems odd to build an instance of a class and set properties, just to call a single method - when we can just as easily pass them in as parameters to a static (C#/Java terminology) method. It's not like we *want* something with state here - we're not going to do anything else afterwards with the combination of the zone and period.

Of course, such a static method could create a stateful object itself if it want to, but I wouldn't have that as the public API - unless I wanted to implement an interface or something similar. (I can imagine creating a "transition formatter" or something similar...)

Just MHO though, and definitely coming from a position of ignorance in Ruby idioms. Maybe creating instances like this is the norm there...